addons/mail/controllers/thread.py

```diff
@@ -212,3 +212,19 @@ def _can_edit_message(cls, message, **kwargs):
     @classmethod
     def _can_delete_attachment(cls, message, **kwargs):
         return cls._can_edit_message(message, **kwargs)
+
+    @http.route("/mail/thread/unsubscribe", methods=["POST"], type="jsonrpc", auth="user")
+    def mail_thread_unsubscribe(self, res_model, res_id, partner_ids):
+        thread = self.env[res_model].browse(res_id)
+        thread.message_unsubscribe(partner_ids)
+        return Store(
+            thread, [], as_thread=True, request_list=["followers", "suggestedRecipients"]
+        ).get_result()
+
+    @http.route("/mail/thread/subscribe", methods=["POST"], type="jsonrpc", auth="user")
+    def mail_thread_subscribe(self, res_model, res_id, partner_ids):
+        thread = self.env[res_model].browse(res_id)
+        thread.message_subscribe(partner_ids)
+        return Store(
+            thread, [], as_thread=True, request_list=["followers", "suggestedRecipients"]
+        ).get_result()
```

Comment:
Now that I think about it, why do we add these routes and return it by rpc rather than sending the data on the bus for the current user? in the existing methods.

Reply:
> Now that I think about it, why do we add these routes and return it by rpc rather than sending the data on the bus for the current user? in the existing methods.

Don't add bus code in message_(un)subscribe, they are optimized to avoid useless noise. Keep higher level methods when bus is really required.